cpp/src/TimerThread.cpp

```diff
@@ -0,0 +1,158 @@
+//-----------------------------------------------------------------------------
+//
+//	TimerThread.cpp
+//
+//  Timer for scheduling future events
+//
+//	Copyright (c) 2017 h3ctrl <h3ctrl@gmail.com>
+//
+//	SOFTWARE NOTICE AND LICENSE
+//
+//	This file is part of OpenZWave.
+//
+//	OpenZWave is free software: you can redistribute it and/or modify
+//	it under the terms of the GNU Lesser General Public License as published
+//	by the Free Software Foundation, either version 3 of the License,
+//	or (at your option) any later version.
+//
+//	OpenZWave is distributed in the hope that it will be useful,
+//	but WITHOUT ANY WARRANTY; without even the implied warranty of
+//	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//	GNU Lesser General Public License for more details.
+//
+//	You should have received a copy of the GNU Lesser General Public License
+//	along with OpenZWave.  If not, see <http://www.gnu.org/licenses/>.
+//
+//-----------------------------------------------------------------------------
+
+#include "TimerThread.h"
+#include "platform/Log.h"
+
+using namespace OpenZWave;
+
+//-----------------------------------------------------------------------------
+// <TimerThread::TimerThreadEntryPoint>
+// Main entry point for the timer thread.
+//-----------------------------------------------------------------------------
+void TimerThread::TimerThreadEntryPoint
+(
+		Event* _exitEvent,
+		void* _context
+)
+{
+	TimerThread* timer = (TimerThread*)_context;
+	if( timer )
+	{
+		timer->TimerThreadProc( _exitEvent );
+	}
+}
+
+//-----------------------------------------------------------------------------
+// <TimerThread::TimerThread>
+// Constructor.
+//-----------------------------------------------------------------------------
+TimerThread::TimerThread
+(
+):
+m_timerEvent( new Event() ),
+m_timerMutex( new Mutex() ),
+m_timerTimeout( Wait::Timeout_Infinite )
+{
+}
+
+//-----------------------------------------------------------------------------
+// <TimerThread::~TimerThread>
+// Destructor.
+//-----------------------------------------------------------------------------
+TimerThread::~TimerThread
+(
+)
+{
+	m_timerMutex->Lock();
+	for ( list<TimerEventEntry *>::iterator it = m_timerEventList.begin(); it != m_timerEventList.end(); ++it ) {
+		delete (*it);
+	}
+	m_timerMutex->Unlock();
+	m_timerMutex->Release();
```

Comment:
Could you use the Scoped Locking Macro's we use elsewhere for this. (LockGuard - defined in Util.h)